src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -1123,27 +1123,20 @@ bool FullyConnected::useSparseWeightsDecompression() {
     return true;
 }
 
-void FullyConnected::fuseDecompressionMultiply(const NodePtr& constData) {
-    fuseDecompressionConstant(constData, decompressionMultiply);
+void FullyConnected::fuseDecompressionMultiplyPtr(const NodePtr& constData) {
+    fuseDecompressionConstantPtr(constData, decompressionMultiplyPtr);
 }
 
-void FullyConnected::fuseDecompressionSubtract(const NodePtr& constData) {
-    fuseDecompressionConstant(constData, decompressionSubtract);
+void FullyConnected::fuseDecompressionSubtractPtr(const NodePtr& constData) {
+    fuseDecompressionConstantPtr(constData, decompressionSubtractPtr);
 }
 
-void FullyConnected::fuseDecompressionConstant(const NodePtr& constData, std::vector<float>& decompressionValues) {
+void FullyConnected::fuseDecompressionConstantPtr(const NodePtr& constData, MemoryCPtr& decompressionValuesPtr) {
     auto *constInputNode = dynamic_cast<node::Input *>(constData.get());
     if (!constInputNode) {
         IE_THROW() << "Cannot cast " << constData->getName() << " to Input";
     }
-    auto constBlob = constInputNode->getMemoryPtr();
-    const auto elementsCount = constBlob->getDescWithType<BlockedMemoryDesc>()->getPaddedElementsCount();
-    decompressionValues.resize(elementsCount);
-    cpu_convert(constBlob->getData(),
```

Comment:
Don't we need precision convertion anymore? Looking on the test zero-point is expressed as low precision constant.

Reply:
You are right: we still need conversion, but only if constant precision is not equal to decompression precision (FP32 for the moment). I added this logic
However, as I remember, we planned to store nf4 decompression constants as is, without conversion. So this logic should be changed when nf4 support on oneDNN side is added